### PR TITLE
pending changes workflow

### PR DESCRIPTION
**Pull request by ssickles**
This PR adds and updates all the endpoints associated with our new workspace changes workflow.

### CODE REVIEW

File: types/workspace_change.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const (
+	ChangeActionAdd    ChangeAction = "add"
+	ChangeActionUpdate ChangeAction = "update"
+	ChangeActionDelete ChangeAction = "delete"
+)
+
+type WorkspaceChange struct {
+	tableName    struct{}     `pg:"workspace_changes"`
+	Id           int64        `json:"id" pg:"id,pk"`
+	WorkspaceUid uuid.UUID    `json:"workspaceUid" pg:"workspace_uid"`
```
Comment (BSick7) [inline types/workspace_change.go:32]: Drop the pg struct tags from this library.

File: env_variables.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+func (ev EnvVariables) Create(stackId, blockId, envId int64, input []types.EnvVariableInput) (*types.WorkspaceChangeset, error) {
+	raw, _ := json.Marshal(input)
+	res, err := ev.Client.Do(http.MethodPost, ev.basePath(stackId, blockId, envId), nil, nil, json.RawMessage(raw))
+	if err != nil {
+		return nil, err
+	}
+
+	var changeset *types.WorkspaceChangeset
+	if err := response.ReadJson(res, changeset); response.IsNotFoundError(err) {
```
Comment (BSick7) [inline env_variables.go:31]: You can swap out the entire if block for the generic method

```suggestion
	return response.ReadJsonPtr[types.Workspace Changeset](res, changeset)
```

**Review by BSick7 — APPROVED** [review]
Left a couple of suggestions.

File: types/workspace_change.go (hunk 1 above)
Comment (ssickles) [reply types/workspace_change.go:32]: Done

File: env_variables.go (hunk 2 above)
Comment (ssickles) [reply env_variables.go]: Done

**Pull request merged**